Frontend/src/pages/registration_edit/components/registrationEditor.jsx

```diff
@@ -1,27 +1,21 @@
 import { EventSelector } from '@thewca/wca-components'
 import React, { useEffect, useState } from 'react'
 import { Button, Checkbox, TextArea } from 'semantic-ui-react'
-import getCompetitionInfo from '../../../api/competition/get/get_competition_info'
+import { useCompetitorInfo, useHeldEvents } from '../../../api/helper/hooks'
 import { getSingleRegistration } from '../../../api/registration/get/get_registrations'
 import { updateRegistration } from '../../../api/registration/patch/update_registration'
-import getCompetitorInfo from '../../../api/user/get/get_user_info'
 import { setMessage } from '../../../ui/events/messages'
 import LoadingMessage from '../../../ui/loadingMessage'
 import styles from './editor.module.scss'
 
 export default function RegistrationEditor({ user_id, competition_id }) {
-  const [competitorInfo, setCompetitorInfo] = useState({})
-  const [heldEvents, setHeldEvents] = useState([])
+  const { eventsLoading, heldEvents } = useHeldEvents(competition_id)
```

Comment:
Since the hook is returning an object with properties `isLoading` and `heldEvents`, `eventsLoading` will always be undefined. You need to either change the hook to return an array `[isLoading, heldEvents]`, or change this const to `{ isLoading: eventsLoading, heldEvenst }`.

Reply:
oops, changed